### PR TITLE
Merge baseplate

### PR DESCRIPTION
**Pull request by Berkyle**
# Code changes
Merge baseplate changes into master, which includes:

1. Eslint changes
   * Kept all eslint changes but removed the import aliasing until we can get craco working .
2. Snackbar dependency
3. Login modal (we've decided to move forward with custom authentication)
4. AppBar and Navigation Drawer
   * This will require a bit more attention between the two of us as we've both contributed good shit for the Drawer / Appbar and we want to keep the best of each implementation
   * I also like the distinction between NavList vs UtilityList
   * (done)
5. Speaking of directory structure, kept your changes with renaming all directories to lowercase
   * This was a fucking nightmare, git apparently does not track changes to directory case by default.
   * I updated my git configuration to `git config core.ignorecase false` which made it much easier.
6. Removed package-lock.json, proptypes, rainbow band
7. Pulled in all context providers to a single component in `src/contexts/index.tsx`
8. Broke out styles into their own component-specific files

# Future work (could do it now but this PR is large enough as is):
1. Need to add a PR template.
2. Add some documentation utility (discussed using GH pages, TSDoc, etc)
3. Create `_jss` subdirectories
4. Move all hooks to `src/hooks` directory; organize hooks by functionality
5. Move jsx in `src/contexts` to `src/components/common` or something to try to keep contexts only concerned with business
6. Update the "Report Bugs" to a modal instead of a GitHub issue (we could set up a cloud function pretty easily to just create a GH issue for us but that's not a priority atm)
7. Using named imports (ie import from `@material-ui/core` instead of `@material-ui/core/AppBar`

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -52,7 +55,10 @@
     "eslint": "^6.8.0",
     "eslint-config-prettier": "^6.9.0",
     "eslint-config-react-app": "^5.1.0",
+    "eslint-import-resolver-custom-alias": "^1.2.0",
```
Comment (juzerzarif) [inline package.json]: Don't need this resolver anymore

File: src/Hooks/useStyles.tsx [hunk 2]
```diff
@@ -1,4 +1,4 @@
-import { makeStyles, createStyles } from '@material-ui/core/styles';
+import { createStyles, makeStyles } from '@material-ui/core/styles';
```
Comment (juzerzarif) [inline src/Hooks/useStyles.tsx]: Are any of the styles in this file being used? If they are they should be broken out and put adjacent to the components

File: src/hooks/useStyles.tsx [hunk 3]
```diff
@@ -0,0 +1,90 @@
+import { createStyles, makeStyles } from '@material-ui/core/styles';
```
Comment (juzerzarif) [inline src/hooks/useStyles.tsx]: duplicate file

**Review by juzerzarif — COMMENTED** [review]
Don't worry about merging the two headers and navs together. There will have to be another PR with the nav changes anyway. This PR should solely focus on merging all the code in baseplate into master with no actual code changes.

File: src/hooks/useStyles.tsx (hunk 3 above)
Comment (Berkyle) [reply src/hooks/useStyles.tsx]: yeah, having some issues with git not picking up changes in casing for directory names. will fix

Comment (Berkyle) [reply src/hooks/useStyles.tsx]: okay got it that was a nightmare

**Pull request merged**